cpp/src/arrow/sparse_tensor.cc

```diff
@@ -500,13 +668,45 @@ Status MakeSparseTensorFromTensor(const Tensor& tensor,
     case SparseTensorFormat::CSC:
       return MakeSparseTensorFromTensor<SparseCSCIndex>(tensor, index_value_type, pool,
                                                         out_sparse_index, out_data);
+    case SparseTensorFormat::CSF:
+      return MakeSparseTensorFromTensor<SparseCSFIndex>(tensor, index_value_type, pool,
+                                                        out_sparse_index, out_data);
+
     // LCOV_EXCL_START: ignore program failure
     default:
       return Status::Invalid("Invalid sparse tensor format");
       // LCOV_EXCL_STOP
   }
 }
 
+namespace {
+
+template <typename TYPE, typename IndexValueType>
+void ExpandSparseCSFTensorValues(int64_t dimension, int64_t offset, int64_t first_ptr,
+                                 int64_t last_ptr, const SparseCSFIndex* sparse_index,
+                                 const TYPE* raw_data, const std::vector<int64_t> strides,
+                                 const std::vector<int64_t> axis_order, TYPE* out) {
```

Comment:
Pass vectors as `const&`.

Reply:
Got it. Would it make sense to make `int64_t`s `const` too?

Reply:
No, they are just values.